RelayTestApp/Assets/Scripts/brainCloud/BrainCloudManager.cs

```diff
@@ -0,0 +1,443 @@
+
+#region RTT functions
+
+    //Getting input from other members
+    void OnRelayMessage(short netId, byte[] jsonResponse)
+    {
+        var memberProfileId = m_bcWrapper.RelayService.GetProfileIdForNetId(netId);
+        string jsonMessage = Encoding.ASCII.GetString(jsonResponse);
+        var json = JsonReader.Deserialize<Dictionary<string, object>>(jsonMessage);
+        Lobby lobby = StateManager.Instance.CurrentLobby;
+        foreach (var member in lobby.Members)
+        {
+            if (member.ID == memberProfileId)
+            {
+                var op = json["op"] as string;
+                if (op == "move")
+                {
+                    var data = json["data"] as Dictionary<string, object>;
+
+                    member.IsAlive = true;
+                    member.MousePosition.x = (int)data["x"];
+                    member.MousePosition.y = -(int)data["y"] + _mouseYOffset;
+                }
+                else if (op == "shockwave")
+                {
+                    var data = json["data"] as Dictionary<string, object>;
+                    Vector2 position = new Vector2((int) data["x"], -(int) data["y"]);
+                    member.ShockwavePositions.Add(position);
+                }
+                break;
+            }
+        }
+    }
+
+    // We received a lobby event through RTT
+    void OnLobbyEvent(string jsonResponse)
+    {
+        Dictionary<string, object> response = JsonReader.Deserialize<Dictionary<string, object>>(jsonResponse);
+        Dictionary<string, object> jsonData = response["data"] as Dictionary<string, object>;
+
+        // If there is a lobby object present in the message, update our lobby
+        // state with it.
+        if (jsonData.ContainsKey("lobby"))
+        {
+            StateManager.Instance.CurrentLobby = new Lobby(jsonData["lobby"] as Dictionary<string, object>,
+                jsonData["lobbyId"] as string);
+            //If we're still in lobby, then update the list of users
+            if (StateManager.Instance.CurrentGameState == GameStates.Lobby)
+            {
+                GameManager.Instance.UpdateLobbyState();
+                StateManager.Instance.isLoading = false;
+            }
+            
+        }
+        //Using the key "operation" to determine what state the lobby is in
+        if (response.ContainsKey("operation"))
+        {
+            var operation = response["operation"] as string;
+            switch (operation)
+            {
+                case "DISBANDED":
+                {
+                    var reason = jsonData["reason"] as Dictionary<string, object>;
+                    if ((int) reason["code"] != ReasonCodes.RTT_ROOM_READY)
+                    {
+                        // Disbanded for any other reason than ROOM_READY, means we failed to launch the game.
+                        CloseGame(true);
+                    }
+
+                    break;
+                }
+                case "STARTING":
```

Comment:
If you have a loading screen, here is the moment to show it. While waiting for the "ROOM_READY".
🚲